### ISSUE TITLE
Cannot install through Cygwin due to another strange error message

### ISSUE THREAD

**Issue by DandelionSprout** [funilrys/PyFunceble#124]
## Bug description
While trying to reinstall PyFunceble 3.3.3 from scratch through "Pure Python method" in Cygwin, trying to run `python3 setup.py test`, `python3 setup.py install`, or `PyFunceble -v`, leads to this error: 

```
imree@ImresPC /cygdrive/c/Users/imree/Desktop/Skrivebordsbilder/PyFunceble-greier/PyFunceble
$ python3.7 setup.py test
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2742, in requires
    deps.extend(dm[safe_extra(ext)])
KeyError: 'all'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 168, in <module>
    "pyfunceble=PyFunceble.cli:tool",
  File "/usr/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.7/site-packages/setuptools/command/test.py", line 226, in run
    installed_dists = self.install_dists(self.distribution)
  File "/usr/lib/python3.7/site-packages/setuptools/command/test.py", line 209, in install_dists
    ir_d = dist.fetch_build_eggs(dist.install_requires)
  File "/usr/lib/python3.7/site-packages/setuptools/dist.py", line 698, in fetch_build_eggs
    replace_conflicting=True,
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 795, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2745, in requires
    "%s has no such extra feature %r" % (self, ext)
pkg_resources.UnknownExtra: python-box 4.0.4 has no such extra feature 'all'
```

## Modifications under `.PyFunceble.yaml`

None that I'm aware of.

## Reproduction

1) Delete all folders on the PC that contain the string "PyFunceble".
2) Open Cygwin.
3) Do the initial steps of "Pure Python method", which are `git clone https://github.com/funilrys/PyFunceble.git` and `cd PyFunceble`.
4) Try to run `python3.7 setup.py test`.
5) Receive the error seen above.
6) Go to Cygwin's update tool and reinstall all `python37(...)` packages that are already installed.
7) Try to run `python3.7 setup.py test` again.
8) Receive the error seen above again.

## Expected behavior
PyFunceble works.

## Versions

**OS:** Windows 10 May 2020 Update + Cygwin 3.1.7
**Python Version:** 3.7.7
**PyFunceble Version:** 3.3.3

## Additional context
Despite the error claiming that `/usr/lib/python3.7/site-packages/pkg_resources/__init__.py` is where the error comes from, it appears to me to actually be from `C:\cygwin64\lib\python3.7\site-packages\pkg_resources\__init__.py`.

**Comment by DandelionSprout** [comment]
Okay, so it turned out just 3min later that `python-box` was not supposed to be v4.0.4, but actually v5.1.1, and that `python-box had to be installed through pip3 and not Cygwin's package manager. So nevermind. It's seemingly all fixed now. 😅

**Comment by spirillen** [comment]
Hey @DandelionSprout 

You should leave this one open, as no version have been specified in the requirement file

https://github.com/funilrys/PyFunceble/blob/c8bca124c40858fd3586145a129c5c2f1c9a962e/requirements.txt#L8

This is something we should have a eye on, Would you please reopen the issue?

**Comment by DandelionSprout** [comment]
Ah, okay. Sure thing.

**Comment by spirillen** [comment]
Thanks :smile:

**Comment by funilrys** [comment]
Thanks a lot for the feedback @DandelionSprout! 

-----

Fixed with https://github.com/funilrys/PyFunceble/commit/f8afe4db4f50db6f3dd9ecabdafe81c5ef6c6902.

**P.S: Help us, DO NOT close this issue manually. This issue is going to be closed automatically once https://github.com/funilrys/PyFunceble/commit/f8afe4db4f50db6f3dd9ecabdafe81c5ef6c6902 will be merged to master.**

**Comment by funilrys** [comment]
Just for info regarding the removed milestone: The fix of this is into the `4.0.0`.  Feel free to test and give us some feedback.

**Comment by funilrys** [comment]
Closing so that I can keep track of what is still to be done. 

Feel free to test and reopen if not fixed or if necessary.
